sos/report/plugins/shmcli.py

```diff
@@ -0,0 +1,109 @@
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support debug data')
+    ]
+
+    def setup(self):
+        cmd = "/opt/dell/ServerHardwareManager/" \
+              "ServerHardwareManagerCLI/bin/shmcli"
+
+        subcmds = [
+            'list adapters',
+            'list physical enclosures',
+            'list failed drives'
+        ]
+
+        for subcmd in subcmds:
+            self.add_cmd_output(
+                "%s %s" % (cmd, subcmd),
+                suggest_filename="shmcli_%s" % (subcmd))
+
+        models = []
+
+        # Get the storage hardware models
+        result = self.exec_cmd('lsscsi -g')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (line.split()[2] == 'DELL'):
+                    models.append(line.split()[3])
+        models = list(set(models))
+
+        subcmds = [
+            'list emms',
+            'list drawers',
+            'list emm slots',
+            'list drive slots',
+            'list fans',
+            'list temp sensors',
+            'list voltage sensors',
+            'list current sensors',
+            'list power supplies',
+            'info enclosure'
+        ]
+
+        result = self.collect_cmd_output(
+            '%s list enclosures' % (cmd),
+            suggest_filename='shmcli_list_enclosures'
+        )
+        if result['status'] == 0:
+            for line in result['output'].splitlines()[2:-2]:
+                _line = line.split()
+                if any(m in _line for m in models):
+                    adapt_index = _line[-1]
+                    enc_index = _line[0]
+                    for subcmd in subcmds:
+                        _cmd = ("%s %s -a=%s -enc=%s"
+                                % (cmd, subcmd, adapt_index, enc_index))
+                        _fname = _cmd.replace(cmd, 'shmcli')
+                        self.add_cmd_output(_cmd, suggest_filename=_fname)
+                    if self.get_option('debug'):
+                        logpath = self.get_cmd_output_path(make=False)
+                        _dcmd = ("%s getdebugcli -a=%s -enc=%s"
+                                 % (cmd, adapt_index, enc_index))
+                        _dname = _dcmd.replace(cmd, 'shmcli')
+                        _odir = (" -outputdir=%s" % (logpath))
+                        self.add_cmd_output(
+                            _dcmd + _odir, suggest_filename=_dname,
+                            timeout=300
+                        )
+
+        result = self.collect_cmd_output(
+            '%s list drives' % (cmd),
+            suggest_filename='shmcli_list_drives'
+        )
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (len(line.split()) > 6):
+                    if (line.split()[0] not in ['WWN', '---']):
+                        _cmd = ("%s info drive -d=%s"
+                                % (cmd, line.split()[0]))
```

Comment:
Calling `line.split()` mutliple times is redundant - isnt it better to store it in a variable for performance reasons?